src/poetry/console/commands/init.py

```diff
@@ -213,6 +223,20 @@ def handle(self) -> int:
             if self.io.is_interactive():
                 self.line("")
 
+        sort_dependencies = self.option("sort-dependencies")
+        question = "Would you like to sort your dependencies alphabetically?"
+        if self.confirm(question, True):
+            sort_dependencies = True
+            if self.io.is_interactive():
+                self.line("")
+        if sort_dependencies is True:
+            requirements = OrderedDict(sorted(requirements.items()))
+            dev_requirements = OrderedDict(sorted(dev_requirements.items()))
+            local_config_file = TOMLFile(Path.cwd() / "poetry.toml")
+            local_config_file_contents = document()
+            local_config_file_contents.update({"dependencies": {"sort": True}})
+            local_config_file.write(local_config_file_contents)
```

Comment:
The dependency sorting in `poetry init` currently works like this:

- For noninteractive use, add a `--sort-dependencies` flag.
- For interactive use, after the user has added all their dependencies, ask them if they want to sort the dependencies alphabetically.
- If dependency sorting is selected (either interactive or noninteractive), Poetry will write a `poetry.toml` file to the new project with `dependencies.sort` set to `true`.

Happy to consider suggestions on the implementation.

Reply:
Hey,

in my opinion we shouldn't introduce another flag or questions here and just respect whatever is set in the config.

Reply:
Sure, that's fine. I reverted the new flag.

By "the config," you mean the user's global Poetry config?

I guess the approach would be to read the global config, and if there's a `dependencies.sort` setting there, use that.